lib/senders/client_name.js

```diff
@@ -0,0 +1,60 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const request = require('request')
+const config = require('../../config').getProperties()
+const log = require('./log')('db')
+
+const OAUTH_SERVER = config.oauth.url
+const clientCache = {}
```

Comment:
The names change rarely so I'm OK with a never-expiring cache here, but I also wonder if there's some drop-in caching package we could use that would add a simple ttl.

Reply:
hm good idea!